File: remote/src/main/java/com/kelsos/mbrc/presenters/BrowseTrackPresenter.java
package com.kelsos.mbrc.presenters;

import com.kelsos.mbrc.annotations.Queue.Action;
import com.kelsos.mbrc.domain.Track;
import com.kelsos.mbrc.ui.views.BrowseTrackView;

public interface BrowseTrackPresenter {
  void bind(BrowseTrackView view);

  void load();

  void queue(Track track, @Action String action);
}

File: app/src/main/kotlin/com/kelsos/mbrc/ui/navigation/library/tracks/BrowseTrackPresenter.kt
package com.kelsos.mbrc.ui.navigation.library.tracks

import com.kelsos.mbrc.annotations.Queue.Action
import com.kelsos.mbrc.domain.Track
import com.kelsos.mbrc.ui.navigation.library.tracks.BrowseTrackView

interface BrowseTrackPresenter {
  fun bind(view: BrowseTrackView)

  fun load()

  fun queue(track: Track, @Action action: String)

  fun load(page: Int, totalItemsCount: Int)
}
